pyincore/analyses/wfnfunctionality/wfnfunctionality.py

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) 2022 University of Illinois and others. All rights reserved.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License v2.0 which accompanies this distribution,
+# and is available at https://www.mozilla.org/en-US/MPL/2.0/
+
+import copy
+import numpy as np
+import pandas as pd
+import networkx as nx
+
+from pyincore import BaseAnalysis, NetworkDataset
+from pyincore.analyses.wfnfunctionality.wfnfunctionalityutil import WfnFunctionalityUtil
+
+
+class WfnFunctionality(BaseAnalysis):
+    """Computes water facility network functionality.
+
+    Args:
+        incore_client: Service client with authentication info
+    """
+
+    def __init__(self, incore_client):
+        super(WfnFunctionality, self).__init__(incore_client)
+
+    def run(self):
+        """Execute water facility network functionality analysis """
+
+        # Obtain tank nodes
+        tank_nodes = self.get_parameter('tank_node_list')
+
+        # Obtain pump station nodes
+        pumpstation_nodes = self.get_parameter('pumpstation_node_list')
+
+        # Get network dataset
+        network_dataset = NetworkDataset.from_dataset(self.get_input_dataset('wfn_network'))
+        edges_wfl_gdf = network_dataset.links.get_dataframe_from_shapefile()
```

Comment:
This is not the exact line but in tornadoepndamage analysis, it does the validation of the network dataset. You might want to put it somewhere in here. The code is in tornadoepndamage.py's line 126

Reply:
Done